### PR TITLE
Last used Db file is no longer suggested if it doesnt exist, First time use update

### PR DESCRIPTION
**Pull request by xxshadowfang**
This PR addresses to things

1. The Login screen will no longer suggest files that don't exist

2. The first time user popup only occurs if any preference data does not exist.

### CODE REVIEW

File: src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java [hunk 1]
```diff
@@ -65,22 +65,17 @@ public void initialize(URL fxmlFileLocation, ResourceBundle resources) {
 
         try {
             String lastDBFile = PreferenceHandler.getStringPreference(PreferenceHandler.LAST_DATABASE_FILE_KEY);
-            if (lastDBFile != null)
-                setChosenFile(new File(lastDBFile));
+            if (lastDBFile != null) {
+                File file = new File(lastDBFile);
+                if(file.exists()&&!file.isDirectory()){
```
Comment (LadyMopar) [inline src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java]: Don't forget about formatting the files

File: src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java [hunk 2]
```diff
@@ -65,22 +65,17 @@ public void initialize(URL fxmlFileLocation, ResourceBundle resources) {
...
+                    this.setChosenFile(file);
+                }
+            } else{
+                setUpIntroHelp();
+            }
         } catch (Exception e) {
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/window/LoginPageController.java]: What is this catching?

**Comment by shellajt** [comment]
How can we remove our preference data to test this?

**Comment by xxshadowfang** [comment]
User directory>.java>.userprefs> then delete the Ledgersoftware folder. That's it on Linux not sure what the windows is.

**Comment by taylerhow** [comment]
So, if the user deletes their last used DB, but still has other DBs in their file system (in their user directory, for instance), then the new user popup will appear because we aren't searching a directory anymore?

**Comment by xxshadowfang** [comment]
@taylerhow Nope. even if the last file doesnt exist any more there is still preference data so the new dbfile wont pop up

**Pull request merged**